### DISCUSSION TITLE
vehicle falls off and penetrates the map, what's the problem?

### DISCUSSION THREAD

**Discussion by phoenixsfly** [carla-simulator/carla#5442] [Q&A]
I try to spwan a vehicle, but it falls off and penetrates the map, what's the problem?
The Map was generated by Roadrunner and ingested in Carla packages by Docker

![image](https://cdn.discordapp.com/attachments/579382818673655828/975592130980511824/CarlaUE4_2022-05-16_10-38-43_00_00_05-00_00_08.gif)

**Comment by shh1v** [comment]
Please attach the code too.

**Comment by phoenixsfly** [comment]
```python3
import carla
import random

client = carla.Client('localhost', 2000)
world = client.get_world()

maplst = client.get_available_maps()
# Load my map
client.load_world(maplst[-1])

# Reload current map and reset state
client.reload_world()


world.get_names_of_all_objects()

vehicle_bps = world.get_blueprint_library().filter('*vehicle*')
vehicle_bp = random.choice(vehicle_bps)
spawn_point = random.choice(world.get_map().get_spawn_points())
ego_vehicle=world.spawn_actor(vehicle_bp, spawn_point)



```

**Comment by SExpert12** [comment]
If you go by random choice then it would happen. You have to choose location and spawn the vehicle on that particular location.
